### PR TITLE
[SPARK-26986][ML] Add JAXB reference impl to build for Java 9+

### PR DESCRIPTION
**Pull request by srowen**
## What changes were proposed in this pull request?

Add reference JAXB impl for Java 9+ from Glassfish. Right now it's only apparently necessary in MLlib but can be expanded later.

## How was this patch tested?

Existing tests particularly PMML-related ones, which use JAXB.
This works on Java 11.

### CODE REVIEW

**Comment by vruusmann** [comment]
The JAXB reference implementation is not a monolith, but a collection of many libraries. Depending on the use case - "1) unmarshal XML only, 2) marshal XML only or 3) both marshal and unmarshal XML" - it should be possible to narrow down this collection.

Apache Spark MLlib "implements" the second use case ("marshal XML only"), and there are no plans to implement the other two use cases in the foreseeable future? If so, then it should be possible to eliminate maybe two or three libraries. Can't say which off the top of my head though.

Is this kind of "optimization of incoming dependencies" relevant at all? Ifit is, then maybe I should take some time to experiment and document this stuff (provided that there is no prior art available).

**Comment by srowen** [comment]
I think it would be nice to trim down the new dependencies, yes. Like you I'm not sure which ones are needed for the 'write-only' use case, but this change is limited in scope to MLlib now and so would be reasonable to keep it narrow. If you have suggestions -- is there a better JAXB impl -- would be glad to hear them. That said, these deps aren't big or weird, so it's not a big deal to keep the dependency simple and not try to mess with exclusions.

**Comment by vruusmann** [comment]
I was thinking about compiling an overview in the form of a table: rows - libraries, columns - use cases, cells - one of "required" or "optional".

Possibly, one table for JDK 9 and another one for JDK 11.

This analysis is something that I can use to help my own customers. For example, there's one specific library that should be excluded in order to disable the collection of SAX Locator information doing unmarshalling (to reduce the memory usage). I tell them about this neat trick, then they ask me the name of this library, and I can't remember.

Putting this topic into my blog's TODO list. Might take a week or two to address, which is probably too long regarding this issue.

**Comment by srowen** [comment]
Sounds good. I will merge this to help unblock Java 11 work, but am happy to hear later that we can trim down the dependencies a little.

**Comment by vruusmann** [comment]
@srowen Some observations about building and deploying JPMML-Model plus GlassFish Metro combination on Java SE 8, 9, 10 and 11:
http://openscoring.io/blog/2019/02/28/jpmml_model_api_configuring_jaxb_dependency

TLDR: I would argue that the following three library dependencies are completely unnecessary for Apache Spark's "native PMML exporter" use case:

- `com.sun.xml.fastinfoset:FastInfoset`
- `org.glassfish.jaxb:txw2`
- `org.jvnet.staxex:stax-ex`

Sample declaration:

```xml
<dependency>
	<groupId>org.glassfish.jaxb</groupId>
	<artifactId>jaxb-runtime</artifactId>
	<version>2.3.2</version>
	<exclusions>
		<exclusion>
			<groupId>com.sun.xml.fastinfoset</groupId>
			<artifactId>FastInfoset</artifactId>
		</exclusion>
		<exclusion>
			<groupId>org.glassfish.jaxb</groupId>
			<artifactId>txw2</artifactId>
		</exclusion>
		<exclusion>
			<groupId>org.jvnet.staxex</groupId>
			<artifactId>stax-ex</artifactId>
		</exclusion>
	</exclusions>
</dependency>
```

**Comment by srowen** [comment]
OK sounds good, I can make a follow up to exclude them, for now, with a comment.
